Sources/Commands/SwiftTool.swift

```diff
@@ -594,6 +594,33 @@ public class SwiftTool {
             throw error
         }
     }
+    
+    /// Evaluate extensions for any reachable targets in the graph, and return a mapping from targets to corresponding evaluation results.
+    func evaluateExtensions(graph: PackageGraph) throws -> [ResolvedTarget: [ExtensionEvaluationResult]] {
+        do {
+            // Configure the inputs to the extension evaluation.
+            // FIXME: These paths are still fairly preliminary.
+            let buildEnvironment = try buildParameters().buildEnvironment
+            let dataDir = try self.getActiveWorkspace().dataPath
+            let extensionsDir = dataDir.appending(component: "extensions")
+            let cacheDir = extensionsDir.appending(component: "cache")
+            let extensionRunner = try DefaultExtensionRunner(cacheDir: cacheDir, manifestResources: self._hostToolchain.get().manifestResources)
+            let outputDir = extensionsDir.appending(component: "outputs")
+            // FIXME: Too many assumptions!
+            let execsDir = dataDir.appending(components: try self._hostToolchain.get().triple.tripleString, buildEnvironment.configuration.dirname)
+            let diagnostics = DiagnosticsEngine()
+            
+            // Create the cache directory, if needed.
+            try localFileSystem.createDirectory(cacheDir, recursive: true)
```

Comment:
do we also need to create the outputs dir?

Reply:
That's a good point.  Technically we don't have to create either of them, since that happens lower down in the stack.  We can probably just make it part of the contract that `evaluateExtensions()` creates the directories as needed, so that not every client has to.  I'll fix it.  Thanks!

Reply:
Removed this call as redundant.